### ISSUE TITLE
Add alt text to vignettes

### ISSUE THREAD

**Issue by MichaelChirico** [tidyverse/ggplot2#4256]
The plots in the following vignettes don't have `alt` tags (search `//img[not(@alt)]` in the HTML rendering):

 - `extending-ggplot2`
 - `ggplot2-specs`

I believe the simplest way to fix this is to add a `fig.cap` for all of the chunks producing images (I still haven't landed on the easiest way to replicate `![text caption](path/to/img.png "alt text")` in an R chunk).

**Comment by yutannihilation** [comment]
Thanks for filing this issue. Do you know what kind of texts are usually captioned as alt texts of plots?

(It's shame on me that I didn't know ggplot2 is accessible for blind users thanks to BrailleR...! <https://www.stat.auckland.ac.nz/~paul/Talks/JSM2018/index-theme-white.html>)

**Comment by MichaelChirico** [comment]
I'm afraid I'm also no expert on this :\

Anyone else following this issue please feel free to chime in!

What I've been going for when doing this myself is to try and describe in words the visual takeaway of the plot that was to be conveyed. Very similar to what `BrailleR` appears to do automatically (great find by the way! although it was removed from CRAN, it appears to be [here](https://github.com/ajrgodfrey/BrailleR)) -- we might even just try and get `BrailleR` to generate the plots & then hard-code its output? Though of course that might have licensing implications (`BrailleR` is GPL-2).

Anyway, as an example, here's this plot from `extending-ggplot2`:

![A scatterplot (generated by `geom_point`) with no title, an x-axis labelled "displ" ranging roughly from 2 to 7, and a y-axis labelled "hwy" ranging roughly from 10 to 45. The points display a strong negative correlation; overlaid on the points is a boundary polygon as generated by `stat_chull` which encircles all of the other points.](https://user-images.githubusercontent.com/7606389/98771676-febecc80-23b2-11eb-932a-bee0335f7fb5.png)

I might caption this like (I also added it to the alt in this comment)

> A scatterplot (generated by `geom_point`) with no title, an x-axis labelled "displ" ranging roughly from 2 to 7, and a y-axis labelled "hwy" ranging roughly from 10 to 45. The points display a strong negative correlation; overlaid on the points is a boundary polygon as generated by `stat_chull` which encircles all of the other points.

**Comment by yutannihilation** [comment]
> we might even just try and get `BrailleR` to generate the plots & then hard-code its output? Though of course that might have licensing implications (`BrailleR` is GPL-2).

Thanks, sounds good to me. I'm not confident, but I don't think the licensing of BrailleR matters here. It's different than the cases when we distribute the software itself. For example, we can use the plot created by ggplot2 (or base R plots) with whatever license.

Maybe we can keep this issue open so that some expert will help us :)

**Comment by eliocamp** [comment]
This is a great idea. It seems that for now, the simplest way to add alt text would be to use fig.cap and then add `fig_caption: false` to the vignette yaml. 

```yaml
output: 
  rmarkdown::html_vignette:
    fig_caption: false
```

It works correctly: 

![A scatterplot (generated by geom_point) with no title, an x-axis labelled "displ" ranging roughly from 2 to 7, and a y-axis labelled "hwy" ranging roughly from 10 to 45. The points display a strong negative correlation; overlaid on the points is a boundary polygon as generated by stat_chull which encircles all of the other points.](https://user-images.githubusercontent.com/8617595/104364383-45d15700-54f5-11eb-8f83-06e4c65e023e.png)

Of course, this only works because these vignettes don't need to show captions, but if captions were necessary, then we'd be out of luck. From what I've seen, [this PR](https://github.com/yihui/knitr/pull/1900) should address this limitation of knitr and add a separate alt-text parameter. Let's hope it's merged soon! :crossed_fingers:

**Comment by yutannihilation** [comment]
Just a quick note. As https://github.com/tidyverse/ggplot2/pull/4461 adds support for BrailleR, I tried if it works on pkgdown site, but it seems it cannot be done with just installing the package. Maybe we also need to tweak something on R Markdown's side?

diff: https://github.com/tidyverse/ggplot2/compare/master...yutannihilation:doc/install-BrailleR-pkgdown-site-test
result: https://yutannihilation.github.io/ggplot2/dev/articles/extending-ggplot2.html

**Comment by teunbrand** [comment]
Closing this as fixed by #5226
